### PR TITLE
Adds async event creation via rabbit.

### PR DESCRIPTION
**Pull request by justinlittman**
refs https://github.com/sul-dlss/preservation_catalog/issues/1860

## Why was this change made? 🤔
Non-blocking goodness.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

**Comment by jcoyne** [comment]
Can we discuss whether this should be in the same library or a new library?  It seems like this new functionality might be completely disjoint from the old synch stuff.

**Comment by justinlittman** [comment]
Definitely can discuss. My inclination would be to keep in the same library to minimize overhead/changes, until it proves to actually be disjoint and then split.

**Comment by jcoyne** [comment]
@justinlittman preservation_catalog has no other use for DSC other than the events service. I think that's a pretty good indication of disjunction.

File: lib/dor/services/client.rb [hunk 1]
```diff
@@ -126,41 +126,36 @@ def configure(url:, token:, enable_get_retries: false)
...
-
-          # @note when token & token_header are nil, this line is required else
-          #   the Faraday instance will be passed an empty block, which
-          #   causes the adapter not to be set. Thus, everything breaks.
-          builder.adapter Faraday.default_adapter
-          builder.headers[:user_agent] = user_agent
-          builder.headers[TOKEN_HEADER] = "Bearer #{token}"
-          builder.request :retry, max: 4, interval: 1, backoff_factor: 2 if with_retries
-        end
+        # @connection ||= ConnectionWrapper.new(connection: build_connection, get_connection: build_connection(with_retries: enable_get_retries))
```
Comment (jcoyne) [inline lib/dor/services/client.rb:152]: Cruft?

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Comment by ndushay** [comment]
this PR needs to be rebased.

**Pull request closed without merging**